cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/ScheduleTaskRunner.java

```diff
@@ -136,26 +113,21 @@ public ScheduleTaskRunner(Store store, ProgramRuntimeService runtimeService, Pre
    *
    * @return a {@link ListenableFuture} object that completes when the program completes
    */
-  private ListenableFuture<?> execute(final Program program, ProgramOptions options)
-    throws TaskExecutionException {
-    ProgramRuntimeService.RuntimeInfo runtimeInfo = runtimeService.run(program, options);
+  private ListenableFuture<?> execute(final Id.Program id, final ProgramType type, Map<String, String> sysArgs,
+                                      Map<String, String> userArgs) throws IOException, TaskExecutionException {
+    ProgramRuntimeService.RuntimeInfo runtimeInfo;
+    try {
+      runtimeInfo = lifecycleService.start(id, type, sysArgs, userArgs, false);
+    } catch (ProgramNotFoundException e) {
+      throw new TaskExecutionException(UserMessages.getMessage(UserErrors.PROGRAM_NOT_FOUND), e, false);
```

Comment:
think you should just have the method throw ProgramNotFoundException instead of converting it to TaskExecutionException.  Seems like they are two very different types of errors.